src/instruction.rs

```diff
@@ -57,6 +57,62 @@ impl fmt::Display for ArithmeticOperator {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum LogicalOperand {
+    LiteralInteger(i64),
+    MemoryReference(MemoryReference),
+}
+
+impl fmt::Display for LogicalOperand {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match &self {
+            LogicalOperand::LiteralInteger(value) => write!(f, "{}", value),
+            LogicalOperand::MemoryReference(value) => write!(f, "{}", value),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum LogicalOperator {
+    Binary(BinaryLogic),
+    Unary(UnaryLogic),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum BinaryLogic {
+    And,
+    Or,
```

Comment:
The [grammar](https://github.com/quil-lang/quil/blob/master/grammars/Quil.g4#L172-L178) marks `Or` as deprecated (presumably in favor of `Ior` and `Xor`). Should we still include it?

Reply:
ditto, above 😄

Reply:
IMO, no. The [live version of the spec](https://quil-lang.github.io/) doesn't include it (or the others).

Reply:
I have removed all deprecated commands